src/client/Client.cc

```diff
@@ -9921,6 +9935,13 @@ int64_t Client::_read(Fh *f, int64_t offset, uint64_t size, bufferlist *bl)
   lat = ceph_clock_now();
   lat -= start;
   logger->tinc(l_c_read, lat);
+  if(conf->client_maintain_atime && is_atime_need_update(in)){
+    in->atime = ceph_clock_now();
+    in->change_attr++;
+    in->mark_caps_dirty(CEPH_CAP_FILE_EXCL);
```

Comment:
Clients will generally not have this cap when reading, so this code is not safe. See also `Client::_setattr` which updates mtime/ctime.

The correct way to do this I think is let the kernel drive atime updates via setattr calls. Have you tried turning that on via FUSE?

@jtlayton anything to add?

Reply:
I have tested that read in ceph will not modify atime. If you only read files, atime will never change.In some scenarios, we may need to modify atime after reading, instead of being unchanged, so we modified it with reference to the rules of vfs .